gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinTaskRunner.java

```diff
@@ -544,23 +544,25 @@ void connectHelixManagerWithRetry() {
   private void addInstanceTags() {
     HelixManager receiverManager = getReceiverManager();
     if (receiverManager.isConnected()) {
-      // The helix instance associated with this container should be consistent on helix tag
-      List<String> existedTags = receiverManager.getClusterManagmentTool()
-          .getInstanceConfig(this.clusterName, this.helixInstanceName).getTags();
-      Set<String> desiredTags = new HashSet<>(
-          ConfigUtils.getStringList(this.clusterConfig, GobblinClusterConfigurationKeys.HELIX_INSTANCE_TAGS_KEY));
-      if (!desiredTags.isEmpty()) {
-        // Remove tag assignments for the current Helix instance from a previous run
-        for (String tag : existedTags) {
-          if (!desiredTags.contains(tag))
-            receiverManager.getClusterManagmentTool().removeInstanceTag(this.clusterName, this.helixInstanceName, tag);
-          logger.info("Removed unrelated helix tag {} for instance {}", tag, this.helixInstanceName);
+      try {
+        Set<String> desiredTags = new HashSet<>(ConfigUtils.getStringList(this.clusterConfig, GobblinClusterConfigurationKeys.HELIX_INSTANCE_TAGS_KEY));
+        if (!desiredTags.isEmpty()) {
+          // The helix instance associated with this container should be consistent on helix tag
+          List<String> existedTags =
+              receiverManager.getClusterManagmentTool().getInstanceConfig(this.clusterName, this.helixInstanceName).getTags();
```

Comment:
Why does moving this line here address the issue? 

In what situations does a config not exist for an instance? Seems to me like the sequence of events is:
```
on start(), connectHelixManager() makes taskrunner join as a participant
Helix should make an instance config for the taskrunner
```

In what scenario does the above not happen?

Reply:
It seems to me that this would return empty (but clearly it doesn't since it causes an exception).

Reply:
I believe there are some yarn related assumptions (on container start registering the instance config or something) and basically we are not setting that up for all use cases. I saw a `setInstanceConfig` method in `ZKHelixAdmin` that should be called explicitly in the non-Yarn case perhaps if Helix is not automatically creating one if taskrunner joins. From testing the code, previously we never entered and tested this code block since `desiredTags` is empty for our use case. This shouldn't change the streaming logic because the `existedTags` definition is only used inside the code block so I've moved the declaration to where it's used and caught the exception that can be thrown.

Reply:
Correct. The current implementation doesn't affect streaming logic and should work. I am pointing out that a missing instance config is an indication of a larger problem. i.e. I don't think this change addresses the core issue which is that we are missing an instance config which causes the exception. 

```
instance [instance] does not exist in cluster [cluster]
```

Reply:
It seems like Helix does not automatically register the participant as an instance config if it joins as a participant. We may need to call `setInstanceConfig` and double check our assumptions.